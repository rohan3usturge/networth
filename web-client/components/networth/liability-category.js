import { Liability } from "./liability";

const LiablityCategory = ({ category, onChange }) => {
  const handleChange = (changedItem) => {
    const changed = { ...category };
    const newItems = category.items.map((item) => {
      if (item.lineItem.id === changedItem.lineItem.id) {
        return changedItem;
      }
      return item;
    });
    changed.items = newItems;
    if (onChange) {
      onChange(changed);
    }
  };

  return (
    <div className="mb-3">
      <div className="h6">{category.meta.name}</div>
      <ul className="list-group list-group-flush">
        {category.items.map((item) => {
          return (
            <Liability key={item.id} liability={item} onChange={handleChange} />
          );
        })}
      </ul>
    </div>
  );
};

export { LiablityCategory };
