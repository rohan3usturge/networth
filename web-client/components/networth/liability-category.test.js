import { shallow } from "enzyme";
import { LiablityCategory } from "./liability-category";

describe("AssetCategory", () => {
  it("should render properly and match snapshot", () => {
    const wrapper = shallow(
      <LiablityCategory
        category={{ meta: {}, items: [] }}
        currency="CAD"
        disabled={true}
        onChange={() => {}}
        key="2323"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
