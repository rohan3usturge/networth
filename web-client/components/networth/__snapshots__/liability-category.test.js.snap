// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetCategory should render properly and match snapshot 1`] = `
<Wrapper
  category={
    Object {
      "items": Array [],
      "meta": Object {},
    }
  }
  currency="CAD"
  disabled={true}
  i18n={Object {}}
  onChange={[Function]}
  t={[Function]}
  tReady={false}
/>
`;
